egs/swbd/s5c/local/nnet3/multi_condition/run_ivector_common.sh

```diff
@@ -0,0 +1,148 @@
+      steps/compute_cmvn_stats.sh data/${datadir}_tmp exp/make_mfcc/${datadir}_tmp $mfccdir || exit 1;
+      utils/fix_data_dir.sh data/${datadir}_tmp
+
+      utils/copy_data_dir.sh --spk-prefix sp1.0- --utt-prefix sp1.0- data/${datadir} data/temp0
+      utils/combine_data.sh data/${datadir}_sp data/${datadir}_tmp data/temp0
+      utils/fix_data_dir.sh data/${datadir}_sp
+      rm -r data/temp0 data/${datadir}_tmp
+    done
+  fi
+
+
+  if [ $stage -le 2 ]; then
+    #obtain the alignment of the perturbed data
+    steps/align_fmllr.sh --nj 100 --cmd "$train_cmd" \
+      data/${input_data_dir}_sp data/lang_nosp exp/tri4 exp/tri4_ali_nodup_sp || exit 1
+  fi
+
+  clean_data_dir=${input_data_dir}_sp
+else
+  clean_data_dir=${input_data_dir}
+fi
+
+
+if [ $stage -le 3 ]; then
+  if [ ! -d "simulated_rirs_8k" ]; then
+    # Download the simulated RIR package with 8k sampling rate
+    wget --no-check-certificate http://www.openslr.org/resources/26/sim_rir_8k.zip
+    unzip sim_rir_8k.zip
+  fi
+
+  # corrupt the data to generate reverberated data 
+  # this script modifies wav.scp to include the reverberation commands, the real computation will be done at the feature extraction
+  # if --include-original-data is true, the original data will be mixed with its reverberated copies
+  python steps/data/reverberate_data_dir.py \
+    --prefix "rev" \
+    --rir-set-parameters "0.3, simulated_rirs_8k/smallroom/rir_list" \
+    --rir-set-parameters "0.3, simulated_rirs_8k/mediumroom/rir_list" \
+    --rir-set-parameters "0.3, simulated_rirs_8k/largeroom/rir_list" \
+    --speech-rvb-probability 1 \
+    --num-replications $num_data_reps \
+    --source-sampling-rate 8000 \
+    --include-original-data true \
+    data/${clean_data_dir} data/${clean_data_dir}_rvb${num_data_reps}
+fi
+
+
+if [ $stage -le 4 ]; then
+  mfccdir=mfcc_rvb
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $mfccdir/storage ]; then
+    date=$(date +'%m_%d_%H_%M')
+    utils/create_split_dir.pl /export/b0{1,2,3,4}/$USER/kaldi-data/egs/swbd-$date/s5b/$mfccdir/storage $mfccdir/storage
+  fi
+
+  for dataset in ${clean_data_dir}_rvb${num_data_reps}; do
+    utils/copy_data_dir.sh data/$dataset data/${dataset}_hires
+
+    # do volume-perturbation on the training data prior to extracting hires
+    # features; this helps make trained nnets more invariant to test data volume.
+    utils/data/perturb_data_dir_volume.sh data/${dataset}_hires
+
+    steps/make_mfcc.sh --nj 70 --mfcc-config conf/mfcc_hires.conf \
+        --cmd "$train_cmd" data/${dataset}_hires exp/make_hires/$dataset $mfccdir;
+    steps/compute_cmvn_stats.sh data/${dataset}_hires exp/make_hires/${dataset} $mfccdir;
+    utils/fix_data_dir.sh data/${dataset}_hires;
+  done
+fi
+
+
+# ivector extractor training
+if [ $stage -le 5 ]; then
+  steps/train_lda_mllt.sh --cmd "$train_cmd" --num-iters 13 \
+    --splice-opts "--left-context=3 --right-context=3" \
+    5500 90000 data/train_100k_nodup_hires \
```

Comment:
@vijayaditya just train the lda_mllt transform on clean data is good enough, and this can avoid copying the alignment.